guides/managing-current-user.md

```diff
@@ -64,14 +64,57 @@ export default Ember.Service.extend({
 
   load() {
     if (this.get('session.isAuthenticated')) {
-      return this.get('store').find('user', 'me').then((user) => {
+      return this.get('store').findRecord('user', 'me').then((user) => {
         this.set('user', user);
       });
     }
   }
 });
 ```
 
+#### Using a dedicated endpoint and returning actual user id in response
+
+```js
+// app/services/current-user.js
+import Ember from 'ember';
+
+const { inject: { service }, isEmpty, RSVP } = Ember;
+
+export default Ember.Service.extend({
+  session: service('session'),
+  store: service(),
+
+  load() {
+    if (this.get('session.isAuthenticated')) {
+      return this.get('store').queryRecord('user', { me: true }).then((user) => {
+        this.set('user', user);
+      });
+    }
+  }
+});
+
+// app/adapters/user.js
+import ApplicationAdapter from './application';
+
+export default ApplicationAdapter.extend({
+  urlForQueryRecord(query) {
+    if (query.me) {
+      delete query.me;
+      return `${this._super(...arguments)}/me`;
+    }
+
+    return this._super(...arguments);
+  }
+});
+```
+
+Note: Using `store.queryRecord` is the correct whey to query for a record when id is
+unknown. Ember data expects the returned model to have the same id, as otherwise an
+unused empty record with `id` of `me` is in the store.
+
+The adapter override will generate `api/users/me` when `store.queryRecord` is
+invoked with a query param where the `me` param is present.
```

Comment:
We should put the explanation before the code sample so the guide is easier to follow when reading through it top-down.